### PR TITLE
Replace wmic with Powershell cmdlets

### LINKED ISSUE #246: wmic is deprecated, use PowerShell cmdlets instead
**Is your feature request related to a problem? Please describe.**
`wmic` is deprecated.

**Describe the solution you'd like**
Package should use PowerShell cmdlets to get printers on Windows instead of `wmic`.

**Describe alternatives you've considered**
None.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by ferretwithaberet**
**Changes**
- Replace wmic with Powershell cmdlets on Windows
- Change tests to adapt to changes
- Remove yarn.lock as this project uses npm

Fixes #246.

### CODE REVIEW

**Comment by artiebits** [comment]
Hi @ferretwithaberet thank you for your work and support! I appreciate it, and sorry for delays in response

File: src/win32/get-printers.js [hunk 1]
```diff
@@ -1,27 +1,48 @@
...
+      .split(/(\r?\n){2,}/)
+      .map((printer) => printer.trim())
+      .filter((printer) => !!printer)
+      .forEach((printer) => {
+        const printerData = {
+          deviceId: null,
+          name: null,
+        };
+
+        const isValid = printer.split(/\r?\n/).some((line) => {
```
Comment (artiebits) [inline src/win32/get-printers.js:19]: `isValid` is duplicated in `get-default-printer.js`. Could you please move this function to `utils` folder?

File: yarn.lock [hunk 2]
```diff
@@ -1,6590 +0,0 @@
-# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (artiebits) [inline yarn.lock:1]: could you add `yarn.lock` to the `.gitignore` then, please?

File: src/win32/get-printers.js (hunk 1 above)
Comment (ferretwithaberet) [reply src/win32/get-printers.js]: Fixed in e6e7c06.

File: yarn.lock (hunk 2 above)
Comment (ferretwithaberet) [reply yarn.lock:1]: Fixed in e8b5fcf.

**Comment by artiebits** [comment]
thank you

**Comment by artiebits** [comment]
i've published a beta version `3.0.0-beta.1`. you might want to test your changes.

**Comment by ferretwithaberet** [comment]
> i've published a beta version `3.0.0-beta.1`. you might want to test your changes.

Sorry for the late reply. I was going to test it when I had time. I do test my changes while I work on them, I also try to write tests that will cover my changes. I do not own a physical printer so I cannot test it extensively, but I test it with virtual printers. Even if I test it, I cannot think of all the possible cases which can happen so we might have to release it and fix the bugs as we find them(Like it happened with #263). This PR itself, in my opinion, should be safe to publish.

**Comment by artiebits** [comment]
hi @ferretwithaberet! same for me, I don't own a physical printer anymore, so it's hard to test changes :/. i've just published a new version, 3.0.0. thank you again for your contribution, I appreciate it!

**Pull request merged**